### ISSUE TITLE
Feature request: size filter

### ISSUE THREAD

**Issue by risicle** [cachix/cachix#377]
Personally, I almost never want to push a store path that's >1GiB. Most of these end up being massive nonfree binaries that have just been downloaded. Accidentally pushing up copies of the 3GiB `cudatoolkit` wastes everyone's time, bandwidth and storage allowance.

**Comment by domenkozar** [comment]
That sounds more like a feature on the server side, so that it's enforced at one place. Do you expect it to error out?

**Comment by risicle** [comment]
No, I think the client should just ignore it. And it could even be based on pre-compressed size and that would allow it to avoid compressing a massive file only to throw the result away.
